gnovm/pkg/gnolang/types.go

```diff
@@ -2098,6 +2108,44 @@ func assertSameTypes(lt, rt Type) {
 	}
 }
 
+func isEqualityTypes(lt, rt Type) bool {
+	debugPP.Printf("check isEqualityTypes, lt: %v, rt: %v, isLeftDataByte: %v, isRightDataByte: %v \n", lt, rt, isDataByte(lt), isDataByte(rt))
+	if lpt, ok := lt.(*PointerType); ok { // refer to 0f31
```

Comment:
this feels wrong; for why would we need to do it only once, when there are also pointer pointers and so on? 
(but this doesn't matter with TypeID() == TypeID())

Reply:
fixed by your solution.

Reply:
this logic has been moved to another PR #1890.